perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -1013,10 +1015,19 @@ sub process_report : Private {
 }
 
 sub contacts_to_bodies : Private {
-    my ($self, $c, $category) = @_;
+    my ($self, $c, $category, $options) = @_;
 
     my @contacts = grep { $_->category eq $category } @{$c->stash->{contacts}};
 
+    # check that the front end has not indicated that we should not send to a
+    # body. This is usually because the asset code thinks it's not near enough
+    # to a road.
+    if ($options->{do_not_send}) {
+        my %do_not_send_check = map { $_ => 1 } @{$options->{do_not_send}};
+        my @contacts_filtered = grep { !$do_not_send_check{$_->body->id} } @contacts;
+        @contacts = @contacts_filtered if scalar @contacts_filtered;
+    }
```

Comment:
if we've filtered out all the contacts why would want to leave the list of contacts as it was before?

Reply:
You wrote this bit ;) I guess didn't want a report to not get sent anywhere that hadn't told the user this?

Reply:
That was weeks ago, I've changed ;)

I see the point, yes. In theory if it's not going to be sent at all the front end shouldn't let them submit and if it has then it's probably a sign of something up in which case sending things is probably a better option.